### PR TITLE
Update profilePage to include timezone

### PR DESCRIPTION
**Pull request by NikkiWines**
### Details

PR does a couple of things: 

- Removes the now unnecessary `fetchTimezone()` function and updates the `personalDetails`onyx key so that each user's personalDetails also contains a timezone. 
- Updates the `priorityMode` onyx key to comply with how we want to save NVP onyx keys. 
- Updates the selectable area for opening the details page to be the avatar and the user's name (as requested [here](https://github.com/Expensify/Expensify.cash/pull/1593#issuecomment-796798141)
- Updates the Details page so that it now includes an abbreviated timezone name for clarity (ex. 10:51 AM PST instead of just 10:51 AM)

### Fixed Issues
<Please replace GH_LINK with the link to the GitHub issue this Pull Request is fixing>
Fixes https://github.com/Expensify/Expensify/issues/156248

### Tests
For testing you're going to need two accounts: 
`Account A`: This an account who on a policy with multiple different users 
`Account B`: Someone who is on the same policy as account A. 

1. Log into E.com with Account A
2. Navigate to Settings > Account > Preferences > Timezone and modify your timezone 
3. Log into E.cash with Account A and check your `myPersonalDetails` onyx key to confirm it contains the correct timezone. 
4. In E.cash, send a message to Account B
5. Confirm the message has the correct timestamp for your timezone. 
6. Click on the name of Account B in the chat header (not on the avatar) and confirm it opens up the `Details` page. 
7. Confirm you see Account B's timezone in the details. 
8. In E.cash, navigate to Settings > Preferences and confirm the page loads as expected
9. Confirm you can modify your priority mode selection and that the `nvp_priorityMode` onyx key updates correctly. 

### Tested On
- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

#### Web
Screen Shot 2021-03-19 at 10 00 04 AM

#### Desktop
Screen Shot 2021-03-19 at 10 26 03 AM

#### Video of Clicking on Chat Header

https://user-images.githubusercontent.com/16747822/111819334-7048e300-889d-11eb-94a3-306e3bef9a13.mov

### CODE REVIEW

File: src/ONYXKEYS.js [hunk 1]
```diff
@@ -53,8 +50,12 @@ export default {
     BETAS: 'betas',
 
     // NVP keys
+    // Contains the user's payPalMe address
     NVP_PAYPAL_ME_ADDRESS: 'nvp_paypalMeAddress',
 
+    // Contains the user preference for the LHN priority mode
+    NVP_PRIORITY_MODE: 'nvp_priorityMode',
```
Comment (marcaaron) [inline src/ONYXKEYS.js:57]: We might want to add a migration for this. Should be pretty simple copy/paste of the "rename" that we did here

https://github.com/Expensify/Expensify.cash/blob/master/src/libs/migrations/RenameActiveClientsKey.js

Comment (marcaaron) [reply src/ONYXKEYS.js:57]: But we could also skip it since we are still in beta and the worst case scenario is that people will just have to reset their priority mode settings.

File: src/libs/actions/PersonalDetails.js [hunk 2]
```diff
@@ -125,7 +112,7 @@ function fetch() {
     })
         .then((data) => {
             const allPersonalDetails = formatPersonalDetails(data.personalDetailsList);
-            Onyx.merge(ONYXKEYS.PERSONAL_DETAILS, allPersonalDetails);
+            Onyx.set(ONYXKEYS.PERSONAL_DETAILS, allPersonalDetails);
```
Comment (marcaaron) [inline src/libs/actions/PersonalDetails.js:115]: Why the change from `merge()` to `set()`?

Comment (marcaaron) [reply src/libs/actions/PersonalDetails.js:115]: I think there may be some personal details created that are not in the `personalDetailsList` e.g. those added when loading reports or creating a chat with a brand new user. Is there a chance we are overwriting those here?

File: src/libs/actions/PersonalDetails.js [hunk 3]
```diff
@@ -135,7 +122,7 @@ function fetch() {
             myPersonalDetails.lastName = lodashGet(data.personalDetailsList, [currentUserEmail, 'lastName'], '');
 
             // Set my personal details so they can be easily accessed and subscribed to on their own key
-            Onyx.merge(ONYXKEYS.MY_PERSONAL_DETAILS, myPersonalDetails);
+            Onyx.set(ONYXKEYS.MY_PERSONAL_DETAILS, myPersonalDetails);
```
Comment (marcaaron) [inline src/libs/actions/PersonalDetails.js:125]: Similar question about why set and not merge. But I can't think of any place we are modifying the `myPersonalDetails` I think probably after the timezone changes in this PR we are just always overwriting the details with those from the server on page refresh. Which seems fine?

**Review by marcaaron — COMMENTED** [review]
Changes look good! I just have a few questions about the set/merge changes.

File: src/ONYXKEYS.js (hunk 1 above)
Comment (NikkiWines) [reply src/ONYXKEYS.js:57]: yeah I wasn't sure if it was really significant enough to require a change since it wouldn't break the UI like the `timezone` change did. However, happy to add it!

File: src/libs/actions/PersonalDetails.js (hunk 2 above)
Comment (NikkiWines) [reply src/libs/actions/PersonalDetails.js:115]: Before we needed to `merge` otherwise I believe we'd overwrite the `timeZone` that was added to the personalDetails when we called `fetchTimezone()` but now the details returned from `personalDetailsList` and `PersonalDetails_GetForEmails` both contain the timeZone so there's no need to merge instead of set. 

> Is there a chance we are overwriting those here?

I suppose if we're getting different persoanlDetaials from the report, then yes we would be overwriting it, so it may be best to stick with `merge` over `set` in the end.

Comment (marcaaron) [reply src/libs/actions/PersonalDetails.js:115]: Ok cool. I think that's probably a safe bet here. It sort of makes sense to favor using `merge()` over `set()` unless there is a specific reason why we might want to blow away everything else we've got.

File: src/ONYXKEYS.js (hunk 1 above)
Comment (marcaaron) [reply src/ONYXKEYS.js:57]: Kind of a NAB to me and I'd almost learn towards not doing it. Maybe it's good to do just to get into the habit since we'd probably want to avoid this change if the app was widely adopted.

**Comment by marcaaron** [comment]
Can't seem to get the timezone stuff to work (even after updating Auth). I don't have a great idea about why that would be but here's what I tried:

- Clean make on Auth then make again
- Change timezone for someone in e.com
- Refresh page in e.cash and look at that guy's profile page
- Timezone is still wrong

Screen Shot 2021-03-19 at 9 21 28 AM
Screen Shot 2021-03-19 at 9 21 57 AM

logged out response of `personalDetailsList` and a lot of stuff is just empty

Screen Shot 2021-03-19 at 9 39 32 AM

Any ideas?

**Comment by NikkiWines** [comment]
Hmm, I'm not able to reproduce that issue. 

Pre changes: 
Screen Shot 2021-03-19 at 1 31 04 PM
Screen Shot 2021-03-19 at 1 30 41 PM

Post changes:

Screen Shot 2021-03-19 at 1 31 19 PM

**Comment by marcaaron** [comment]
Ok after some back and forth @NikkiWines  figured out that I must be testing with an account that did not have a personalDetails nvp set. We don't create one until the user modifies their avatar or name. 

Maybe we can create a follow up issue to do something about that since it's a little confusing that we aren't picking up the tz in E.cash unless the user has modified some unrelated field first.

**Review by marcaaron — COMMENTED** [review]
Tests good!

**Comment by NikkiWines** [comment]
updated!

**Review by marcaaron — CHANGES_REQUESTED** [review]
Looks great! We have to add the migration to this list here and then we are all set

https://github.com/Expensify/Expensify.cash/blob/546e0f24e05ef5b48ef0df988fa3ef5ccdc54219/src/libs/migrateOnyx.js#L10

**Comment by NikkiWines** [comment]
Oops, my bad forgot that part.

**Comment by NikkiWines** [comment]
updated!

**Comment by NikkiWines** [comment]
updated once more with conflicts resolved.

File: src/libs/migrations/RenamePriorityModeKey.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+                // Fail early here because there is nothing to migrate
+                if (_.isEmpty(oldPriorityMode)) {
+                    console.debug('[Migrate Onyx] Skipped migration RenamePriorityModeKey');
+                    return resolve();
+                }
+
+                Onyx.multiSet({
+                    priorityMode: null,
+                    [ONYXKEYS.NVP_PRIORITY_MODE]: oldPriorityMode,
```
Comment (marcaaron) [inline src/libs/migrations/RenamePriorityModeKey.js:24]: Kinda of unrelated but random thought here is that we should probably not reference `ONYXKEYS` in migration files since the migration is "forever" even tho the constant in Onyx could possibly change. Nothing to do here just a weird thought I guess.

**Review by marcaaron — APPROVED** [review]
Looks great !
